packages/client/src/models/parent.js

```diff
@@ -0,0 +1,126 @@
+import { Db } from '../providers/FirebaseProvider';
+
+const ParentCollectionRef = Db.collection('parents');
+
+/** Firebase Parent Convertor */
+const ParentConverter = {
+    toFirestore: (parent) => {
+        const mentees = [];
+
+        // must convert Mentee object to plain json
+        parent.mentees.forEach(s => {
+            mentees.push({
+                name: s.name,
+                email: s.email,
+                grade: s.grade,
+                subjects: s.subjects
+            });
+        });
+
+        return {
+            name: parent.name,
+            email: parent.email,
+            timezone: parent.timezone,
+            mentees: mentees
+        };
+    },
+
+    fromFirestore: (snapshot, options) => {
+        const data = snapshot.data(options);
+
+        const mentees = [];
+
+        data.mentees.forEach(s => {
+            mentees.push(new Mentee(s.name, s.email, s.grade, s.subjects))
+        });
+
+        return new Parent(
+            data.name,
+            data.email,
+            data.timezone,
+            mentees
+        );
+    }
+}
+
+// this is mainly just a convenient constructor
+export class Mentee {
+    constructor(name, email, grade, subjects) {
+        this.name = name;
+        this.email = email;
+        this.grade = grade;
+        this.subjects = subjects;
+    }
+}
+
+/** Firebase Parent Object */
+export default class Parent {
+    constructor(name, email, timezone, mentees) {
+        this.id = undefined;
+        this.name = name;
+        this.email = email;
+        this.timezone = timezone;
+        this.mentees = mentees;
+
+        this.validate();
+    }
+
+    validate() {
+        if (!this.name) throw Error('Parent must have a name');
+        if (!this.email) throw Error('Parent must have an email');
+        if (!this.timezone) throw Error('Parent must have timezone');
+    }
+
+    /**
+     * Updates an existing parent record in firebase.
+     * Note: the id is populated by firebase. DO NOT mutate
+     *
+     * @return {Promise<void>} a promise indicating successful update.
+     */
+    update() {
+        this.validate();
+
+        if (!this.id) {
+            return Promise.reject('Parent update failed: not initialized with firebase uid');
+        }
+
+        return ParentCollectionRef.doc(this.id)
+            .withConverter(ParentConverter)
+            .update(this);
+    }
+
+    /**
+     * Publishes the current parent instance to firebase. If a parent
+     * with the firebase uid already exists the record is updated.
+     * @param {string} the firebase auth uid. DO NOT use any other value.
```

Comment:
thoughts on passing in `auth.user` instead? this would allow us to avoid having to check if the user is logged in first when calling `create`, we can throw some custom exception inside `create` instead

Reply:
This is a good idea.